### ISSUE TITLE
VirtualBox VM fails to boot after initial install

### ISSUE THREAD

**Issue by blalor** [elasticdog/packer-arch#49]
I'm trying to build a VirtualBox Vagrant box with the latest master (`2016.08.01` Arch ISO) with VirtualBox 5.1.2 on OS X 10.11.5.  Packer is able to create the VM from the ISO (`install-base.sh` completes), but the resulting VM image fails to boot in order to run the shell provisioner.

If I checkout d93e04754a4a5f4c7b40d7935dae8797151b200c and run `packer build -only=virtualbox-iso arch-template.json`, I end up with a working vagrant box.  So it appears there's something up with the August Arch release.  Reporting that here seems like the first logical step. :-)

![screen shot 2016-08-02 at 6 42 07 am](https://cloud.githubusercontent.com/assets/109915/17325944/56066298-587c-11e6-99a9-dfc804d5a96c.png)

```
[(master) packer-arch]> ./wrapacker 
virtualbox-iso output will be in this color.

==> virtualbox-iso: Downloading or copying ISO
    virtualbox-iso: Downloading or copying: https://mirrors.kernel.org/archlinux/iso/latest/archlinux-2016.08.01-dual.iso
==> virtualbox-iso: Starting HTTP server on port 8704
==> virtualbox-iso: Creating virtual machine...
==> virtualbox-iso: Creating hard drive...
==> virtualbox-iso: Creating forwarded port mapping for communicator (SSH, WinRM, etc) (host port 3205)
==> virtualbox-iso: Starting the virtual machine...
==> virtualbox-iso: Waiting 5s for boot...
==> virtualbox-iso: Typing the boot command...
==> virtualbox-iso: Waiting for SSH to become available...
```

**Comment by tomswartz07** [comment]
I'm seeing this too.

Perhaps we revert the latest ISO update change?

**Comment by jps3** [comment]
+1 on this for me as well. For both VMware Fusion and Virtualbox. 

I'm not seeing anything obvious in the packer setup here, and am wondering if it's a side-effect of one of the `pacman` installs? (I'll try to dig in further to see if I can help out, perhaps.)

**Comment by cosmo0920** [comment]
I've created a PR to revert August 2016 ISO change.

**Comment by tomswartz07** [comment]
Ok. Let's revert it for now until we figure out what's going on with the latest install media.

We'll keep this Issue open until we get it figured out. :)

**Comment by tomswartz07** [comment]
It seems that the issue is related to the way that Syslinux is installed.
The new ldlinux modules were changed on how they load as of Syslinux 5, and became 'mandatory' as of Sylinux 6. 
http://www.syslinux.org/wiki/index.php?title=Library_modules#All_Syslinux_variants_need_an_additional_ldlinux_module

I'm looking for a tweak on the install-base.sh script that will help fix it.

**Comment by elasticdog** [comment]
@tomswartz07 thank you for diving into this! I just got back from vacation and am still catching up on work.

**Comment by tomswartz07** [comment]
I narrowed the issue down to a change of defaults in 64bit filesystems.

It seems that mkfs.ext4 as of version 1.43 e2fsprogs defaults the filesystem to 64 bit, which breaks support for Syslinux.
http://www.syslinux.org/wiki/index.php?title=Filesystem#ext

I'm pushing a fix shortly.

**Comment by tomswartz07** [comment]
Fixed via a7ff9f2e78e27ce32f09d7416d9c52f351770918
